.github/actions/setup-go/action.yml

```diff
@@ -0,0 +1,53 @@
+name: Setup Go
+description: Setup Golang with efficient caching 
+inputs:
+  only-modules:
+    description: Set to 'true' to only cache modules
+    default: 'false'
+  cache-version: 
+    description: Set this to cache bust
+    default: "1"
+
+runs:
+  using: composite
+  steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: "go.mod"
+          cache: false
+
+      - name: Get branch name
+        if: ${{ inputs.only-modules == 'false' }}
+        id: branch-name
+        uses: tj-actions/branch-names@09ab61130975078eb7cde103fe8d2ae1649a1853 #v6.1
+
+      - name: Set go cache keys
+        shell: bash
+        id: go-cache-dir
+        run: |
+          echo "::set-output name=gomodcache::$(go env GOMODCACHE)" 
+          echo "::set-output name=gobuildcache::$(go env GOCACHE)" 
+
+      - uses: actions/cache@v3
+        name: Cache Go Modules
+        with:
+          path: |
+            ${{ steps.go-cache-dir.outputs.gomodcache }}
+          # The lifetime of go modules is much higher than the build outputs, so we increase cache efficiency
+          # here by not having the primary key contain the branch name
+          key: ${{ runner.os }}-gomod-${{ inputs.cache-version }}-${{ hashFiles('**/go.sum') }}
```

Comment:
What does `**/go.sum` match? Is this meant to cover the integration testing module also?

Reply:
It matches any file within the repo that is named `go.sum`

Reply:
OK. nbd but It would be more efficient to keep them separate if you are trying to optimize more.

Reply:
Fixed 85aa429e0af20b35f6af486da71de370a7cb77c8